hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/NameNodeProxiesClient.java

```diff
@@ -349,6 +349,18 @@ public static ClientProtocol createProxyWithAlignmentContext(
       boolean withRetries, AtomicBoolean fallbackToSimpleAuth,
       AlignmentContext alignmentContext)
       throws IOException {
+    if (!conf.getBoolean(HdfsClientConfigKeys.DFS_OBSERVER_READ_ENABLE,
```

Comment:
We probably should do all the IPC side changes and the pure HDFS ones in a separate PR and then have this PR just with the RBF side.

Reply:
Acknowledged. I'm still trying to verify the patch with a real cluster. Once this is done, I can try splitting it up.

Reply:
Hi @goiri I'm going to be away for a few weeks to do the split. If it's a must have, I can do it when I'm back. Or if anybody has bandwidth they can help out.